src/wallet/wallet.cpp

```diff
@@ -938,6 +938,25 @@ void CWallet::LoadToWallet(const CWalletTx& wtxIn)
     }
 }
 
+void CWallet::RemoveOldTransactions()
+{
+    AssertLockHeld(cs_wallet);
+
+    std::vector<CWalletTx> vWtx;
+    for (std::map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it) {
+        CWalletTx tx = it->second;
+        if (!IsFromMe(*tx.tx) && !IsMine(*tx.tx)) {
```

Comment:
I think this is too broad. Suggest adding an additional condition that the to be deleted transaction is to or from the specific watch-only address we're removing.

Reply:
Hi

Reply:
purge_transactions